File: Uni_MS/angular-ui/src/app/models/sport.ts
export interface Sport {
  id?: number;
  uniqueCode?: string;
  name: string;
  code: string;
  description?: string;
  coachName?: string;
  maxParticipants?: number;
  isActive?: boolean;
  createdAt?: string;
  updatedAt?: string;
}

File: Uni_MS/uni_ms/src/main/java/com/badrulamin/University_Management/entity/Sport.java
package com.badrulamin.University_Management.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "sports")
public class Sport extends BaseEntity {

    @NotBlank
    @Column(nullable = false, unique = true)
    private String name;

    @NotBlank
    @Column(nullable = false, unique = true)
    private String code;

    private String description;

    private String coachName;

    private Integer maxParticipants;

    private Boolean isActive = true;
}
